gateway/tests/proxy/test_httpserver.py

```diff
@@ -0,0 +1,144 @@
+"""Tests HTTPServer"""
+import threading
+from http.server import HTTPServer
+import http, http.client, urllib.parse
+import unittest
+import operator
+from proxy.proxy.httpserver import ProxyRequestHandler
+from http.server import ThreadingHTTPServer, BaseHTTPRequestHandler
+
+
+class TestServerThread(threading.Thread):
```

Comment:
Interesting, why we are not getting lint errors 🤔 

Linter should check for docstrings for tests as well 🤔

Reply:
I'll add the docstrings.

Reply:
I moved the proxy out of the gateway and add `tox.ini` for lint and test